backend/ctgov/api/views.py

```diff
@@ -130,40 +160,184 @@ def post(self, request, *args, **kwargs):
+                    eligibility_max_age
+                )
+            )
+        else:
+            pass
+
         if not first:
             first = 0
         if not last:
             last = 100
 
-        search_results = (
-            BasicSearchM.objects.filter(**filters)
-            .all()
-            .values(
+        if metadata_required:
+            search_results_all = (
+                SearchStudies.objects.filter(**filters)
+                .filter(q_title_acronym)
+                .filter(q_outcome_measure)
+                .filter(q_eligibility_age)
+                .filter(q_age_between)
+                .all()
+            )
+
+            search_results = search_results_all.all().values(
                 "status",
                 "brief_title",
                 "nct_id",
                 "condition_name",
                 "intervention_name",
                 "location_name",
             )[first:last]
-        )
 
-        serializer = BasicSearchSerializer(search_results, many=True)
-        return Response(serializer.data, status=status.HTTP_200_OK)
+            results_count = search_results_all.count()
+
+        else:
+            search_results = (
+                SearchStudies.objects.filter(**filters)
+                .filter(q_title_acronym)
+                .filter(q_outcome_measure)
+                .filter(q_eligibility_age)
+                .filter(q_age_between)
+                .all()
+                .values(
+                    "status",
+                    "brief_title",
+                    "nct_id",
+                    "condition_name",
+                    "intervention_name",
+                    "location_name",
+                )[first:last]
+            )
+
+            results_count = "NA"
+
+        serializer = SearchStudiesSerializer(search_results, many=True)
+        return Response(
+            {
+                "metadata": [{"results_count": str(results_count)}],
+                "search_results": serializer.data,
+            },
+            status=status.HTTP_200_OK,
+        )
+        # return Response(serializer.data, status=status.HTTP_200_OK)
 
 
+# Convert passed date parameter to match with db format
 def convert_to_date(datestr):
     date = datetime.strptime(datestr, "%Y-%m-%d")
     return date
 
 
+# Check whether passed date parameter is valid
 def valid_date(datestr):
-    if not datestr:
-        return False
+    status = True
+    try:
+        if datestr:
+            datetime.strptime(str(datestr), "%Y-%m-%d")
+        else:
+            status = False
+    except (ValueError, NameError):
+        status = False
+
+    return status
 
+
+# Check whether passed date age parameter is either integer or float
+def valid_number(agestr):
+    status = True
     try:
-        datetime.strptime(datestr, "%Y-%m-%d")
-    except ValueError:
-        return False
+        if agestr:
+            status = isinstance(float(agestr), float)
```

Comment:
If you convert something to a float and it doesn't throw a ValueError than isinstance(x, float) is guaranteed to be true....

Reply:
try isinstance("100.50", float)

Reply:
That's not the same thing. Obviously a string isn't an instance of a float. But in your code you are converting agestr to a float so if that succeeds then it will be guaranteed to be a float.

Reply:
try isinstance(float("100.50"), float)